cmd/signature-v4.go

```diff
@@ -229,6 +229,12 @@ func doesPresignedSignatureMatch(hashedPayload string, r *http.Request, region s
 		return errCode
 	}
 
+	// Check if the metadata headers are equal with signedheaders
+	errMetaCode := checkMetaHeaders(extractedSignedHeaders, r)
+	if errMetaCode != ErrNone {
+		return errMetaCode
+	}
+
```

Comment:
// extractSignedHeaders extract signed headers from Authorization header
func extractSignedHeaders(signedHeaders []string, r *http.Request) (http.Header, APIErrorCode) {


This function is already doing this what is special about `checkMetaHeaders` ?

Reply:
checkMetaHeaders will check if there is extra metadata that should not be there, as it was mention in https://github.com/minio/minio/issues/16917, if you add extra headers with metadata which was not used to signed the url,  minio will add the metadata to the file while aws s3 will give an error.

Reply:
@harshavardhana It does not check if there are headers in 'r' that are not signed - it checks if the `extractedSignedHeaders` headers are actually present and match, not the other way.